### PR TITLE
Enrich webhook output file

### LINKED ISSUE #1517: Enrich webhook output log with more than the resp.body

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1517

### CODE REVIEW

File: actions/webhook.go [hunk 1]
```diff
@@ -103,6 +104,36 @@ func (w *Webhook) Run(ctx context.Context, record graveler.HookRecord, writer *H
 	return nil
 }
 
+func writeOutput(ctx context.Context, writer *HookOutputWriter, req *http.Request, resp *http.Response, reqBody []byte) error {
```
Comment (nopcoder) [inline actions/webhook.go:107]: https://golang.org/pkg/net/http/httputil
I was thinking we should use dump request and response

File: actions/service_test.go [hunk 2]
```diff
@@ -8,9 +8,12 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/require"
```
Comment (nopcoder) [inline actions/service_test.go:15]: remove empty line

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: actions/service_test.go [hunk 3]
```diff
@@ -155,4 +161,10 @@ hooks:
 	if runResult != nil {
 		t.Errorf("GetRunResult() result=%v, expected nil", runResult)
 	}
+
+	bytes, err := ioutil.ReadAll(outputReader)
```
Comment (nopcoder) [inline actions/service_test.go:165]: need to capture this one in the DoAndReturn - can't assume the reader will be valid at this point

File: actions/service_test.go [hunk 4]
```diff
@@ -155,4 +161,10 @@ hooks:
...
+	require.NoError(t, err)
+	str := string(bytes)
+	rows := strings.Split(str, "\n")
+	require.Greater(t, len(rows), 10)
```
Comment (nopcoder) [inline actions/service_test.go:169]: You can use bytes.Split if you just want to count rows, and prevent the string()

File: actions/webhook.go [hunk 5]
```diff
@@ -103,6 +112,26 @@ func (w *Webhook) Run(ctx context.Context, record graveler.HookRecord, writer *H
...
+	buf.WriteString("\n\nWebhook response:\n")
+	dumpResp, err := httputil.DumpResponse(resp, true)
+	if err != nil {
+		return err
+	}
+	buf.Write(dumpResp)
+
+	buf.WriteString(fmt.Sprintf("\nRequest duration: %s\n", elapsed))
+
+	return writer.OutputWrite(ctx, buf, int64(buf.Len()))
```
Comment (nopcoder) [inline actions/webhook.go:132]:
```suggestion
	var buf bytes.Buffer
	dumpReq, err := httputil.DumpRequestOut(req, true)
	if err != nil {
		buf.WriteString("Error during request dump: " + err.Error() + "\n")
		return err
	}
	_, err = buf.WriteString("Request:\n")
	if err != nil {
		return err
	}
	_, err = buf.Write(dumpReq)
	if err != nil {
		return err
	}

	dumpResp, err := httputil.DumpResponse(resp, true)
	if err != nil {
		return err
	}
	_, err = buf.WriteString("\n\nResponse:\n")
	if err != nil {
		buf.WriteString("Error during response dump: " + err.Error() + "\n")
		return err
	}
	_, err = buf.Write(dumpResp)
	if err != nil {
		return err
	}

	_, err = buf.WriteString("\nRequest duration: " + elapsed.String() + "\n")
	if err != nil {
		return err
	}
	return writer.OutputWrite(ctx, &buf, int64(buf.Len()))
```

File: actions/webhook.go [hunk 6]
```diff
@@ -80,20 +83,26 @@ func (w *Webhook) Run(ctx context.Context, record graveler.HookRecord, writer *H
...
+	elapsed := time.Since(start)
 	if err != nil {
 		return err
 	}
 	defer func() {
 		_ = resp.Body.Close()
 	}()
 
+	// seeking for logging the request
+	if _, err := reqReader.Seek(0, 0); err != nil {
```
Comment (nopcoder) [inline actions/webhook.go:99]: why?

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: actions/webhook.go (hunk 5 above)
Comment (itaiad200) [reply actions/webhook.go:132]: Did something similar, have a look.

File: actions/webhook.go (hunk 6 above)
Comment (itaiad200) [reply actions/webhook.go:99]: Otherwise `httputil.DumpRequestOut` returns an error.

**Review by nopcoder — APPROVED** [review]

**Pull request merged**